components/Village/Add/index.vue

```diff
@@ -0,0 +1,419 @@
+        this.form.latitude.length &&
+        !this.errors.villageId &&
+        !this.errors.villageName &&
+        !this.errors.city &&
+        !this.errors.district &&
+        !this.errors.longitude &&
+        !this.errors.latitude
+      ))
+    }
+  },
+  watch: {
+    show: {
+      handler (val) {
+        this.modalShow = val
+      },
+      immediate: true
+    },
+    'form.id' () {
+      // This pattern for checking user only allow input number and character (.)
+      const formatText = /(?=.*[^\d.])/g
+      if (formatText.test(this.form.id)) {
+        this.form.id = ''
+      }
+      // formatId is pattern for id village maximal 13 chars (10 digit and 3 symbol) => XX.XX.XX.XXXX
+      const formatId = /(\B(?=(\d{2})+(?!\d))(?=.{4}))/g
+      const newId = this.form.id.replaceAll(/[^0-9]/g, '').replace(formatId, '.')
+      this.form.id = newId
+      if (this.form.id.length < 13 && this.form.id.length > 13) {
+        this.errors.villageId = 'Format isian kode wilayah salah'
+      } else {
+        this.errors.villageId = null
+      }
+    },
+    'form.name' () {
+      if (this.form.name.length > 0 && this.form.name.length < 3) {
+        this.errors.villageName = 'Isian nama minimal 3 karakter.'
+      } else if (this.form.name.length > 100) {
+        this.errors.villageName = 'Isian nama maksimal 100 karakter.'
+      } else {
+        this.errors.villageName = null
+      }
+    },
+    'form.city_id' (newId, oldId) {
+      if (newId && newId !== oldId) {
+        this.isDisabledOptionDistricts = false
+        this.form.district_id = ''
+        this.fetchDistricts(newId)
+      } else {
+        this.isDisabledOptionDistricts = true
+      }
+
+      if (newId === undefined) {
+        this.errors.city = 'Isian Kabupaten/Kota wajib diisi'
+      } else {
+        this.errors.city = null
+      }
+    },
+    'form.district_id' () {
+      if (this.form.district_id === undefined) {
+        this.errors.district = 'Isian Kecamatan wajib diisi'
+      } else {
+        this.errors.district = null
+      }
+    },
+    'form.level' () {
+      if (this.form.level === undefined) {
+        this.errors.villageLevel = 'Isian level desa wajib diisi'
+      } else {
+        this.errors.villageLevel = null
+      }
+    },
+    'form.longitude' () {
+      const regexPoint = /^(-?\d+(\.\d+)?)$/
+      if (regexPoint.test(this.form.longitude) || !this.form.latitude.length) {
+        this.errors.longitude = null
+      } else {
+        this.errors.longitude = 'Format isian longitude salah'
+      }
+    },
+    'form.latitude' () {
+      if (this.form.latitude.length === '') {
+        this.form.latitude = ''
+      }
+      const regexPoint = /^(-?\d+(\.\d+)?)$/
+      if (regexPoint.test(this.form.latitude) || !this.form.latitude.length) {
+        this.errors.latitude = null
+      } else {
+        this.errors.latitude = 'Format isian latitude salah'
+      }
+    }
+  },
+  mounted () {
+    this.fetchCity()
+  },
```

Comment:
Ask:

|If not mistaken you're using Nuxt.js right? Is there a reason to fetch data on `mounted` hook instead of using `async fetch`?

Reply:
I don't have any reason for fetching data in mounted, because I usually use mounted in vue. But I just remember in lifecycle hooks nuxt we can fetch data using `fetch` before `mounted` hooks. 

source: [nuxt ](https://nuxtjs.org/docs/concepts/nuxt-lifecycle/) lifecycle hooks https://nuxtjs.org/docs/concepts/nuxt-lifecycle/

Reply:
Yes, if we running universal rendering with Nuxt.js we can use `async fetch` that will run on the server or client, the question is should the `this.fetchCity` be called on the mounted hook or should we move it to `async fetch`?

If you're not running universal mode and only fetch data on the client then `mounted` hook is absolutely fine 👍🏼